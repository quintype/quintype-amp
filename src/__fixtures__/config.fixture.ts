import { Config, PublisherConfig, AMPConfig, OptsTypes } from "../types/config";

export const publisherConfig: PublisherConfig = {
  "cdn-name": "https://thumbor-stg.assettype.com/",
  "text-direction": "ltr",
  "publisher-name": "newslaundry"
};

export const ampConfig: AMPConfig = {
  menu: {
    enabled: true,
    items: [
      {
        title: "Home",
        url: "https://barandbench.com"
      },
      {
        title: "News",
        url: "https://www.barandbench.com/news"
      },
      {
        title: "Dealstreet",
        url: "https://www.barandbench.com/dealstreet"
      },
      {
        title: "Conferences & Paper presentations",
        url: "https://www.barandbench.com/apprentice-lawyer/call-for-papers"
      },
      {
        title: "Recruitment Tracker",
        url: "https://www.barandbench.com/apprentice-lawyer/recruitment-tracker"
      },
      {
        title: "Apprentice Lawyer",
        url: "https://www.barandbench.com/apprentice-lawyer"
      },
      {
        title: "Viewpoint",
        url: "https://www.barandbench.com/view-point"
      },
      {
        title: "Legal Jobs",
        url: "https://www.barandbench.com/legal-jobs"
      }
    ]
  },
  fonts: {
    primary: {
      url: "Open+Sans:300,400,600,700",
      family: '"Open Sans", sans-serif'
    },
    secondary: {
      url: "PT+Serif:400,400italic,700,700italic",
      family: "sans-serif"
    }
  },
  colors: {
    primary: "#294f32",
    secondary: "#763194",
    "footer-background": "#FDBD10",
    "footer-text-color": "#FDBD10",
    "header-background": "#dd0b4d",
    "section-text-color": "#f4e842"
  },
  "logo-url": "https://thumbor-stg.assettype.com/ace/2018-02/f0ba1430-8c21-49a3-a5cf-50c88ce9ec74/martian3.jpg",
  doubleclick: {
    "top-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    },
    "body-ad": {
      width: 300,
      height: 250,
      "unit-path": "/60988533/AMP_Body"
    },
    "bottom-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    }
  },
  "google-client-id-api": false,
  "invalid-elements-strategy": "redirect-to-web-version",
  "google-analytics-tracking-id": "UA-ABCDEFG"
};

export const opts: OptsTypes = {
  slots: {
    story: {
      "top-slot": {
        script: `
          <script async custom-element="amp-date-countdown" src="https://cdn.ampproject.org/v0/amp-date-countdown-0.1.js"></script>
          <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
        `,
        ampHtml: `<amp-date-countdown height="100" class="ts_wrapper" timestamp-seconds="2147483648"
        layout="fixed-height">
        <template type="amp-mustache">
          <div>This is the top slot</div>
          <div>
            {{d}} days, {{h}} hours, {{m}} minutes and {{s}} seconds until <a class="ts_a" href="https://en.wikipedia.org/wiki/Year_2038_problem">Y2K38</a>.
          </div>
        </template>
      </amp-date-countdown>`,
        styles: `.ts_wrapper {
          background-image: linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
          filter: brightness(0.9);
          margin: 15px 0;
          line-height: 50px;
          text-align: center;
          display: block;
          color: white;
        }
        .ts_a {
          color: inherit;
        }`
      },
      "bottom-slot": {
        script: `<script async custom-element="amp-dailymotion" src="https://cdn.ampproject.org/v0/amp-dailymotion-0.1.js"></script>`,
        ampHtml: `<amp-dailymotion data-videoid="x3rdtfy"
          layout="responsive"
          data-ui-highlight="FF4081"
          width="480"
          class="storyTopSlot1"
          height="270">
        </amp-dailymotion>`
      }
    }
  }
};

export const config: Config = {
  publisherConfig,
  ampConfig,
  opts
};
