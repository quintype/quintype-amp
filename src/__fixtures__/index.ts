import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
