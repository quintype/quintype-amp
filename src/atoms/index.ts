import Carousel from "./carousel";
import { Image } from "./image";
import Layout from "./layout";
import Facebook from "./facebook";
import Twitter from "./twitter";
import { Footer } from "./footer";
import { Section } from "./section";
import { Spacer } from "./spacer";
import { Author } from "./author";
import { DateTime } from "./date-time";
import { StoryElement } from "./story-elements/story-element";
import { StoryElements } from "./story-elements";

export {
  Carousel,
  Image,
  Layout,
  Facebook,
  Twitter,
  Footer,
  Section,
  Spacer,
  Author,
  DateTime,
  StoryElement,
  StoryElements
};
