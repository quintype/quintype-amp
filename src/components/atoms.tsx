import React, { Fragment } from 'react'
import { Helmet } from 'react-helmet'
import * as atomTypes from '../type-declarations/atomicComponents'

export const Image = (props: atomTypes.Image) => {
  return <amp-img {...props}></amp-img>
}

export const Title = (props: atomTypes.Title) => {
  return (
    <Helmet>
      <title>{props.children}</title>
    </Helmet>
  )
}

export const Carousel = (props: atomTypes.Carousel) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.2.js"></script>
        <meta name="lorem ipsum" content="carousel script added" />
      </Helmet>
      <amp-carousel {...props}>{props.children}</amp-carousel>
    </Fragment>
  )
}

export const FitText = (props: atomTypes.FitText) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-fit-text" src="https://cdn.ampproject.org/v0/amp-fit-text-0.1.js"></script>
        <meta name="lorem ipsum" content="Fit Text script added" />
      </Helmet>
      <amp-fit-text {...props}>{props.children}</amp-fit-text>
    </Fragment>
  )
}

export const Youtube = (props: atomTypes.Youtube) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
        <meta name="lorem ipsum" content="YOUTUBE script added" />
      </Helmet>
      <amp-youtube {...props}>{props.children}</amp-youtube>
    </Fragment>
  )
}

// const Amp = {Image, Title, Carousel, FitText, Youtube}
// export default Amp