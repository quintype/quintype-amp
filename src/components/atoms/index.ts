import Carousel from './Carousel'
import FitText from './FitText'
import Image from './Image'
import Title from './Title'
import Youtube from './Youtube'

export {Carousel,FitText,Image,Title,Youtube}
