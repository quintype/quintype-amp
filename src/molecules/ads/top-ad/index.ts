import { TopAd } from "./top-ad";
export { TopAd };
