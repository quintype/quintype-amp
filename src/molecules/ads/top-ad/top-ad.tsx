import React from "react";
import { withConfig } from "../../../context";
import { Ad } from "../../../atoms";
import { AdTypes } from "../../../atoms/ad/types";
import { getAdPropsFromConfig, isEmpty, getDataSlotFromUnitPath } from "../helpers";
import { Config } from "../../../types/config";
import styled from "styled-components";

const AdWrapper = styled.div`
  text-align: center;
`;

const TopAdBase = ({ children, config, ...overridingProps }: AdTypes & Config) => {
  const propsForTopAd = isEmpty(overridingProps)
    ? getDataSlotFromUnitPath(getAdPropsFromConfig({ config, adName: "top-ad" }))
    : overridingProps;

  return (
    <AdWrapper>
      <Ad type="doubleclick" {...propsForTopAd}>
        {children}
      </Ad>
    </AdWrapper>
  );
};

export const TopAd = withConfig(TopAdBase);
