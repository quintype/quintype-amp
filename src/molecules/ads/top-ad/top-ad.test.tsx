import React from "react";
import { TopAd } from "./top-ad";
import { mount, shallow } from "enzyme";
import { Layout } from "../../../atoms";
import { textStory, config } from "../../../__fixtures__";
import cloneDeep from "lodash.clonedeep";

const dummyConfig = cloneDeep(config);
dummyConfig.ampConfig.doubleclick["top-ad"] = {
  width: 300,
  height: 250,
  "unit-path": "/4444/lorem/ipsum/sticky"
};
const LayoutWithTopAd = () => (
  <Layout story={textStory} config={dummyConfig}>
    <TopAd />
  </Layout>
);

describe("TopAd", () => {
  it("snapshot test", () => {
    const wrapper = shallow(<TopAd />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render TopAd taking ad props from config", () => {
    const wrapper = mount(<LayoutWithTopAd />);
    expect(wrapper.find("amp-ad").prop("type")).toBe("doubleclick");
    expect(wrapper.find("amp-ad").prop("width")).toBe(300);
    expect(wrapper.find("amp-ad").prop("height")).toBe(250);
    expect(wrapper.find("amp-ad").prop("data-slot")).toBe("/4444/lorem/ipsum/sticky");
  });
  it("should apply overriding props", () => {
    const customPropsForAd = {
      width: "320",
      type: "highlyAnnoyingAd",
      "data-slot": "lorem/ipsum"
    };
    const wrapper = mount(<TopAd {...customPropsForAd} />);
    expect(wrapper.find("amp-ad").prop("data-slot")).toBe("lorem/ipsum");
    expect(wrapper.find("amp-ad").prop("type")).toBe("highlyAnnoyingAd");
    expect(wrapper.find("amp-ad").prop("width")).toBe("320");
  });
});
