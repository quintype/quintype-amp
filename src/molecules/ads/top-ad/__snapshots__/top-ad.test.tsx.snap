// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAd snapshot test 1`] = `
<TopAdBase
  config={Object {}}
/>
`;
